mautrix_telegram/portal/telegram.py

```diff
@@ -613,6 +616,8 @@ async def _backfill_messages(self, source: 'AbstractUser', min_id: Optional[int]
 
     async def handle_telegram_message(self, source: 'AbstractUser', sender: p.Puppet,
                                       evt: Message) -> None:
+        latest_event_timestamp = time()
```

Comment:
```suggestion
        self.latest_event_timestamp = time()
```

That doesn't seem like a global.